server/common/rest.py

```diff
@@ -0,0 +1,205 @@
+from http import HTTPStatus
+import warnings
+from flask import make_response, jsonify
+import copy
+
+from server.common.constants import Axis, DiffExpMode, JSON_NaN_to_num_warning_msg
+from server.common.errors import (
+    FilterError,
+    InteractiveError,
+    JSONEncodingValueError,
+    PrepareError,
+    DisabledFeatureError,
+)
+
+import json
+from server.data_common.fbs.matrix import decode_matrix_fbs
+
+
+def schema_get_helper(data_engine, annotations):
+    """helper function to gather the schema from the data source and annotations"""
+    schema = data_engine.get_schema()
+    # add label obs annotations as needed
+    if annotations is not None:
+        labels = annotations.read_labels(data_engine)
+        if labels is not None and not labels.empty:
+            schema = copy.deepcopy(schema)
+            for col in labels.columns:
+                col_schema = {
+                    "name": col,
+                    "writable": True,
+                }
+                # FIXME: data._get_col_type needs to be refactored.
+                col_schema.update(data_engine._get_col_type(labels[col]))
```

Comment:
I would have thought that `_get_col_type()` would be part of the annotations abstraction, not a data_engine method.

If annotations are abstracted, and separate concerns from the base data format, then the typing of an annotation is not within the data engine domain.

Reply:
Yes, I meant to get back to this.  I refactored how types were figured out between the three use cases (annotations, h5ad and tiledb).